### PR TITLE
Validate the custom address space

### PR DESCRIPTION
**Pull request by ross-p-smith**
# PR for issue #772 

## What is being addressed
When you are creating a workspace and you define an "address_space_size" of `custom`, then this PR will validate that the address space is within the TRE_ADDRESS_SPACE when TRE was originally created.

### CODE REVIEW

File: api_app/services/cidr_service.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from typing import List
 from ipaddress import IPv4Network, NetmaskValueError, collapse_addresses
+from xmlrpc.client import boolean
```
Comment (martinpeck) [inline api_app/services/cidr_service.py:3]: I don't think you want this (see other comment)

File: api_app/services/cidr_service.py [hunk 2]
```diff
@@ -29,6 +24,28 @@ def generate_new_cidr(allocated_subnets: List[str], required_cidr_block_type: in
...
+
+    free_subnets = remove_subnet([allocation_network], core_network)
+
+    for subnet_string in allocated_subnets:
+        free_subnets = remove_subnet(free_subnets, IPv4Network(subnet_string))
+
+    return free_subnets
+
+
+def is_network_available(allocated_subnets: List[str], requested_CIDR) -> boolean:
```
Comment (martinpeck) [inline api_app/services/cidr_service.py:39]: return type should be `bool`

**Review by martinpeck — CHANGES_REQUESTED** [review]
LGTM. Just one small change (use of `bool` instead of the imported type `boolean`)

File: api_app/services/cidr_service.py (hunk 1 above)
Comment (ross-p-smith) [reply api_app/services/cidr_service.py:3]: Do you know - I dont even know where that came from!

**Review by martinpeck — APPROVED** [review]
LGTM

**Pull request merged**